xarray/core/concat.py

```diff
@@ -162,6 +166,8 @@ def concat(
 
         If a callable, it must expect a sequence of ``attrs`` dicts and a context object
         as its only parameters.
+    create_index : bool, default is True
+        Whether to create new PandasIndex objects for any new 1D coordinate variables.
```

Comment:
Consider renaming and rewriting this description? Perhaps `promote_to_index`?

Reply:
How about `create_index_for_new_dim`? Because it's only going to happen when a 0D variable gains a dimension that it did not already have.

Reply:
```
create_index_for_new_dim : bool, default is True
        Whether to create a new ``PandasIndex`` object when the objects being concatenated contain scalar variables named ``dim``.
```

Reply:
That's better!